packages/qdrant-loader-mcp-server/src/qdrant_loader_mcp_server/search/components/vector_search_service.py

```
@@ -135,34 +139,36 @@ async def get_embedding(self, text: str) -> list[float]:
-                raise
+            # Accept either a provider (with .embeddings()) or a direct embeddings client
+            client = (
+                self.embeddings_provider.embeddings()
+                if hasattr(self.embeddings_provider, "embeddings")
+                else self.embeddings_provider
+            )
+            for _ in range(3):
+                try:
+                    vectors = await client.embed([text])
+                    return vectors[0]
+                except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)